src/sparsezoo/v2/integration_validation/validator.py

```diff
@@ -0,0 +1,246 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Validator class that checks, whether the contents of ModelDirectory
+class objects are valid
+"""
+
+import os
+from typing import Callable, Dict, Optional, Set, Tuple, Union
+
+from sparsezoo.v2.directory import Directory
+from sparsezoo.v2.file import File
+from sparsezoo.v2.integration_validation import (
+    validate_cv_classification,
+    validate_cv_detection,
+    validate_cv_segmentation,
+    validate_nlp,
+)
+from sparsezoo.v2.model_directory import ModelDirectory
+
+
+__all__ = ["IntegrationValidator"]
+
+# files/directories that are required for minimal validation
+ESSENTIAL_FILES = {"deployment", "training", "model.onnx"}
+INTEGRATION_NAME_TO_DATA = {
+    "nlp": validate_nlp,
+    "cv_classification": validate_cv_classification,
+    "cv_detection": validate_cv_detection,
+    "cv_segmentation": validate_cv_segmentation,
+}
+
+
+class IntegrationValidator:
+    """
+    Helper class to facilitate the validation of the ModelDirectory class.
+    It is called by the "validate" method of ModelDirectory.
+
+    :param model_directory: ModelDirectory class object to be
+        validated
+    :param essential_files: a minimal set of file names in ModelDirectory,
+        that need to be present, so that the validation is
+        successful
+    :param integration2data: mapping from the integration tag (string) from
+        ModelDirectory's model card to validation function
+    """
+
+    def __init__(
+        self,
+        model_directory: ModelDirectory,
+        essential_files: Set[str] = ESSENTIAL_FILES,
+        integration2data: Dict[str, Callable] = INTEGRATION_NAME_TO_DATA,
```

Comment:
nit: `integration_to_data` also consider different word than data if we're describing a function